### ISSUE TITLE
openssl can't handle \n line endings, needs \r\n 

### ISSUE THREAD

**Issue by cukal** [openssl/openssl#21420]
#### Create a file with a single line: 

`echo "This is a test" > testfile`

#### Generate a signature:
`openssl smime -sign -in testfile -signer yourpem.PEM -inkey yourkey.KEY -outform PEM -out FS.testfile`

#### Verify the signature: 
`openssl smime -verify -in FS.testfile -inform PEM -content testfile -noverify -certfile yourpem.PEM`

output: 
```
This is a test
Verification failure
140104353458064:error:21071065:PKCS7 routines:PKCS7_signatureVerify:digest failure:pk7_doit.c:1108:
140104353458064:error:21075069:PKCS7 routines:PKCS7_verify:signature failure:pk7_smime.c:400:
```

#### Change the line ending: 
`unix2dos testfile`

#### Verify the signature: 
`openssl smime -verify -in FS.testfile -inform PEM -content testfile -noverify -certfile yourpem.PEM`

output: 
```
This is a test
Verification successful
```

Tested on CentOS 7 with: 
`OpenSSL 3.0.8 7 Feb 2023 (Library: OpenSSL 3.0.8 7 Feb 2023)`
`OpenSSL 1.0.2k-fips  26 Jan 2017`

**Comment by t8m** [comment]
One option is to use -binary option with the `openssl smime -sign` command. @DDvO was trying to fix the line ending handling in cms, not sure if smime was also affected by these changes - most probably not.

**Comment by beldmit** [comment]
I think @DDvO 's fix was rolled back by me as it was basically "Let's read the file twice, once for finding out the text/binary mode and the 2nd for the operation".

**Comment by t8m** [comment]
Also have you tried the cms utility? Or that does not work for you at all?

**Comment by cukal** [comment]
@t8m The first development tests using smime (as advised by the remote service integration guide...) were with a fixed file generated on Windows followed by integration tests using files generated on Linux, the verification step failure baffled me so much I started digging for a root cause, finding it gave a clear reason & a quick workaround. 
_High fives and drinks at 3PM._

**Comment by kroeckx** [comment]
Does everything work properly if you use DOS line ending? As far as I know, smime requires it.

**Comment by cukal** [comment]
@kroeckx Yes, adding them fixes it. It seems the S/MIME spec stipulates lines should always end with `<CR><LF>` characters and so does the openssl smime man page underneath `-binary` so we could argue it's not actually a bug as it works when input file adheres to the S/MIME spec, I think we were thrown off course when sign worked but verify didn't.

**Comment by t8m** [comment]
I would still think this is a bug when sign apparently implicitly adds the CRs when it computes the signature, to not add them appropriately as well when verify is called.
